app/org/sagebionetworks/bridge/dynamodb/DynamoFPHSExternalIdentifierDao.java

```diff
@@ -0,0 +1,109 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Resource;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.FPHSExternalIdentifierDao;
+import org.sagebionetworks.bridge.exceptions.EntityAlreadyExistsException;
+import org.sagebionetworks.bridge.exceptions.EntityNotFoundException;
+import org.sagebionetworks.bridge.models.accounts.ExternalIdentifier;
+import org.sagebionetworks.bridge.models.accounts.FPHSExternalIdentifier;
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper.FailedBatch;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
+
+@Component
+public class DynamoFPHSExternalIdentifierDao implements FPHSExternalIdentifierDao {
+
+    private DynamoDBMapper mapper;
+    
+    @Resource(name = "fphsExternalIdDdbMapper")
+    public final void setMapper(DynamoDBMapper mapper) {
+        this.mapper = mapper;
+    }
+    
+    @Override
+    public void verifyExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        // This will throw an exception if the record doesn't exist or has already been registered.
+        getExternalId(externalId, true);
+    }
+
+    @Override
+    public void registerExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        // This will throw an exception if the record doesn't exist or has already been registered.
+        DynamoFPHSExternalIdentifier record = getExternalId(externalId, true);
+        record.setRegistered(true);
+        mapper.save(record);
+    }
+    
+    @Override
+    public void unregisterExternalId(ExternalIdentifier externalId) {
+        checkNotNull(externalId);
+        
+        DynamoFPHSExternalIdentifier record = getExternalId(externalId, false);
+        if (record != null) {
+            record.setRegistered(false);
+            mapper.save(record);
+        }
+    }
```

Comment:
We need unit tests to verify this bug fix. What happens when you unregister an external ID that doesn't exist? What happens when you unregister an external ID that's already registered?